PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/FunctionRunConfigurationProducer.java

```diff
@@ -0,0 +1,114 @@
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.microsoft.intellij.runner.functions;
+
+import com.intellij.codeInsight.MetaAnnotationUtil;
+import com.intellij.execution.Location;
+import com.intellij.execution.RunnerAndConfigurationSettings;
+import com.intellij.execution.actions.ConfigurationContext;
+import com.intellij.execution.actions.LazyRunConfigurationProducer;
+import com.intellij.execution.configurations.ConfigurationFactory;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.util.Comparing;
+import com.intellij.openapi.util.Ref;
+import com.intellij.psi.PsiMethod;
+import com.intellij.util.containers.ContainerUtil;
+import com.microsoft.intellij.runner.AzureRunConfigurationBase;
+import com.microsoft.intellij.runner.functions.core.FunctionUtils;
+import com.microsoft.intellij.runner.functions.localrun.FunctionRunConfiguration;
+import org.jetbrains.annotations.NotNull;
+import org.jsoup.helper.StringUtil;
+
+import java.util.Iterator;
+
+public class FunctionRunConfigurationProducer extends LazyRunConfigurationProducer<AzureRunConfigurationBase> {
+    @NotNull
+    @Override
+    public ConfigurationFactory getConfigurationFactory() {
+        return AzureFunctionSupportConfigurationType.getInstance().getConfigurationFactories()[0];
+    }
+
+    public static boolean isFunctionClassAnnotated(final PsiMethod method) {
+        return MetaAnnotationUtil.isMetaAnnotated(method, ContainerUtil.immutableList(FunctionUtils.AZURE_FUNCTION_ANNOTATION_CLASS));
+    }
+
+    @Override
+    protected boolean setupConfigurationFromContext(AzureRunConfigurationBase configuration, ConfigurationContext context, Ref ref) {
+        if (!(configuration instanceof FunctionRunConfiguration)) {
+            return false;
+        }
+        FunctionRunConfiguration runConfiguration = (FunctionRunConfiguration) configuration;
+        final Location contextLocation = context.getLocation();
+        assert contextLocation != null;
+        Location<PsiMethod> methodLocation = getAzureFunctionMethods(contextLocation);
+        if (methodLocation == null) {
+            return false;
+        }
+        final RunnerAndConfigurationSettings template = context.getRunManager().getConfigurationTemplate(getConfigurationFactory());
+
+        final Module contextModule = context.getModule();
+        final Module predefinedModule = ((FunctionRunConfiguration) template.getConfiguration()).getConfigurationModule().getModule();
```

Comment:
Can we use configuration here directly?

Reply:
For config run from runline bar, there will be no chance for viewing and editing debug settings, use initializeDefaults to set null values like staging folder to a reasonable default value.